python/fluid/gan.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2018 Yi Wang <yi.wang.2005@gmail.com> All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import fluid
+
+
+def create_generator_parameters():
+    return {"Gw": fluid.tensor(dim=[2, 20]), "Gb": fluid.tensor(dim=[20])}
+
+
+def create_discriminator_parameters():
+    return {"Dw": fluid.tensor(dim=[20, 2]), "Db": fluid.tensor(dim=[2])}
+
+
+def generate(noise, params):
+    return fluid.fc(noise, params["Gw"], params["Gb"])
+
+
+with fluid.defun("train_discriminator", mb, params):
+    fluid.optimize(
```

Comment:
I am not sure whether we should combine `backward` and `optimize` together. However, if we want to combine them together, I suggest we use `fluid.optimizers.adam()` or `fluid.optimizers.sgd`, instead of `fluid.optimize`

Reply:
I haven't thought of as much detail. But I feel that we should pack

1. backward, 
1. gradient aggregation, and 
1. parameter update 

in a single function call, so that 

1. we can parallelize them, and
1. we can encapsulate distributed computing details (Kubernetes, MPI, SGE, etc)